spec/support/shared_examples/sign_in.rb

```diff
@@ -324,16 +324,34 @@ def user_with_broken_personal_key(scenario)
     fake_analytics = FakeAnalytics.new
     allow_any_instance_of(ApplicationController).to receive(:analytics).
       and_wrap_original do |original|
-        original_analytics = original.call
-        if original_analytics.request.params[:controller] == 'users/sessions' &&
-           original_analytics.request.params[:action] == 'create'
-          expect(original_analytics.user).to eq(user)
+        if original.receiver.instance_of?(Users::SessionsController) &&
+           original.receiver.action_name == 'create'
+          expect(original.call.user).to eq(user)
```

Comment:
These changes are necessary because the unsuccessful submission will result in a redirect back to the sign-in page, which counterintuitively keeps `original_analytics.request.params[:action] == 'create'` even in the processing of the `#new` action, since the action is processed as a result of the submission's redirect. Since `user_id` is expected to be `nil` on the "Sign in page visited" event, the previous implementation would raise an exception. These changes are to limit assertions to only analytics logged during `#create`.